src/profile-logic/sanitize.js

```diff
@@ -59,19 +62,28 @@ export function sanitizePII(
   const PIIToBeRemoved = maybePIIToBeRemoved;
   const oldThreadIndexToNew: Map<ThreadIndex, ThreadIndex> = new Map();
 
-  let pages;
-  if (profile.pages) {
+  const windowIdToBeSanitized = new Set();
+  let pages = profile.pages;
+  if (pages) {
+    if (PIIToBeRemoved.shouldRemovePrivateBrowsingData) {
+      // slicing here so that we can mutate it later.
+      pages = pages.slice();
+
+      for (let i = pages.length - 1; i >= 0; i--) {
+        const page = pages[i];
+        if (page.isPrivateBrowsing) {
+          windowIdToBeSanitized.add(page.innerWindowID);
+          page.url = removeURLs(page.url, `<Page #${i}>`);
+        }
+      }
+    }
```

Comment:
Mmm I believe I thought of it too. I think I kept it because I thought I'd have to change all references but this was a misunderstanding: I think I thought we referenced the index in the array, but instead we reference the innerWindowID, so I guess we could drop the entry.

Should we also need to make sure that the "tree" of windows is still correct? It _should_ be correct because I guess embedded iframes in a private browsing tab are also private.

Reply:
I added a splice, this seems to work fine.